adapters/audienceNetwork/facebook.go

```diff
@@ -37,267 +41,416 @@ func (a *FacebookAdapter) Name() string {
 	return "audienceNetwork"
 }
 
-// Facebook likes to parallelize to minimize latency
-func (a *FacebookAdapter) SplitAdUnits() bool {
-	return true
-}
-
 func (a *FacebookAdapter) SkipNoCookies() bool {
 	return false
 }
 
-type facebookParams struct {
-	PlacementId string `json:"placementId"`
+type facebookReqExt struct {
+	PlatformId string `json:"platformid"`
+	AuthId     string `json:"authentication_id"`
 }
 
-func coinFlip() bool {
-	return rand.Intn(2) != 0
-}
+func (this *FacebookAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	if len(request.Imp) == 0 {
+		return nil, []error{&errortypes.BadInput{
+			Message: "no impressions provided",
```

Comment:
Consider starting with a capital letter to match the style of the other `BadInput` error.

Reply:
done